### ISSUE TITLE
SAML bundle upgrade

### ISSUE THREAD

**Issue by phavekes** [SURFnet/sp-dashboard#815]
[This issue is imported from pivotal - Originaly created at Jan 16, 2018 by Bart Geesink](https://www.pivotaltracker.com/story/show/154353123)

De SAML bundle wordt voor alle projecten geupgrade

**Comment by phavekes** [comment]
Er gaat nog iets mis:
[2018-01-16 10:02:20] app.ERROR: Could not authenticate, Exception of type "Doctrine\DBAL\Exception\NotNullConstraintViolationException" encountered: "An exception occurred while executing \'INSERT INTO contact (name_id, display_name, email_address) VALUES (?, ?, ?)\' with params [null, "John Doe", "j.doe@example.com"]:  SQLSTATE[23000]: Integrity constraint violation: 1048 Column \'name_id\' cannot be null" [] []
[2018-01-16 10:02:20] request.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\NotNullConstraintViolationException: "An exception occurred while executing \'INSERT INTO contact (name_id, display_name, email_address) VALUES (?, ?, ?)\' with params [null, "John Doe", "j.doe@example.com"]:  SQLSTATE[23000]: Integrity constraint violation: 1048 Column \'name_id\' cannot be null" at /opt/openconext/spdashboard/releases/sp-dashboard-feature_upgrade-stepup-saml-bundle/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 112 {"exception":"[object] (Doctrine\\DBAL\\Exception\\NotNullConstraintViolationException(code: 0): An exception occurred while executing \'INSERT INTO contact (name_id, display_name, email_address) VALUES (?, ?, ?)\' with params [null, \"John Doe\", \"j.doe@example.com\"]:\n\nSQLSTATE[23000]: Integrity constraint violation: 1048 Column \'name_id\' cannot be null at /opt/openconext/spdashboard/releases/sp-dashboard-feature_upgrade-stepup-saml-bundle/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:112, Doctrine\\DBAL\\Driver\\PDOException(code: 23000): SQLSTATE[23000]: Integrity constraint violation: 1048 Column \'name_id\' cannot be null at /opt/openconext/spdashboard/releases/sp-dashboard-feature_upgrade-stepup-saml-bundle/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:107, PDOException(code: 23000): SQLSTATE[23000]: Integrity constraint violation: 1048 Column \'name_id\' cannot be null at /opt/openconext/spdashboard/releases/sp-dashboard-feature_upgrade-stepup-saml-bundle/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:105)"} []
 (Bart Geesink - Jan 16, 2018)

**Comment by phavekes** [comment]
@bartgeesink This problem was caused by the incorrect retrieval of the NameID value from the Saml Assertion. This was fixed in Stepup-saml-bundle version 3.0.0-rc2.

I didn\'t identify this problem in my own functional tests as I performed al my tests as the Admin user. Not triggering the creation of a new SP-dashboard user. 

[I\'ve installed](https://github.com/SURFnet/sp-dashboard/pull/106) this version in the SP-dashboard application. Deploying the feature branch once again should solve the problem. (Michiel Kodde - Jan 16, 2018)
